application_form/services/reservation.py

```diff
@@ -88,11 +103,32 @@ def create_reservation_without_application(
         if user:
             reservation_data["handler"] = user.profile_or_user_full_name
 
+        ownership_type = apartment.project_ownership_type
+        print("ownership_type", ownership_type)
+        if ownership_type.lower() == "haso":
+            existing_reservations = existing_reservations.order_by("queue_position")
+            existing_reservations = existing_reservations.active()
+            right_of_residence_ordering_number = reservation_data[
+                "customer"
+            ].right_of_residence_ordering_number
+            print(
+                "right_of_residence_ordering_number", right_of_residence_ordering_number
+            )
+            max_queue_position = get_queue_position(
+                existing_reservations, right_of_residence_ordering_number
+            )
+            max_list_position = max_queue_position
+            _make_room_for_reservation(
+                apartment_uuid, max_list_position, max_queue_position
+            )
+        else:
+            max_list_position = (max_list_position or 0) + 1
+            max_queue_position = (max_queue_position or 0) + 1
```

Comment:
These variables are now reassigned and they aren't no longer the maximums at this point, but rather the value that will be assigned to the new reservation., I think you should use another variable name for the `+ 1` value, e.g.  `new_list_position` and `new_queue_position`.

```suggestion
            new_list_position = (max_list_position or 0) + 1
            new_queue_position = (max_queue_position or 0) + 1
```